steno/src/model/media/media.cpp

```diff
@@ -0,0 +1,75 @@
+#include "media.h"
+
+#include "model/app_config.h"
+#include "model/stream/video/video_config.h"
+
+#include <QCameraInfo>
+#include <QUrl>
+
+namespace Model
+{
+Media::Media(std::shared_ptr<Config> config)
+    : m_appConfig(config->subConfig(Config::APP))
+    , m_videoConfig(config->subConfig(Config::VIDEO_OUTPUT))
+{
+    initCamera();
+    initRecorder();
+}
+
+void Media::setViewFinder(QCameraViewfinder* view)
+{
+    m_camera->unload();
+    m_camera->setViewfinder(view);
+    m_camera->start();
+}
+
+void Media::startRecorder()
+{
+    m_mediaRecorder->record();
+}
+
+void Media::stopRecorder()
+{
+    m_mediaRecorder->stop();
+}
+
+QMediaRecorder::State Media::recorderState() const
+{
+    return m_mediaRecorder->state();
+}
+
+void Media::initCamera()
+{
+    auto deviceName = m_videoConfig->value(VideoConfig::Key::DEVICE_NAME).toString();
+    for (auto cameraInfo : QCameraInfo::availableCameras())
+    {
+        if (cameraInfo.deviceName() == deviceName)
+        {
+            m_camera.reset(new QCamera(cameraInfo));
+            break;
+        }
```

Comment:
we are not doing this anymore. Consult Walid and the recorder in the current main.